src.ts/wallet/utxohdwallet.ts

```diff
@@ -340,7 +340,27 @@ export class UTXOHDWallet extends BaseWallet {
                 nakedCount = 0;
             }
             derivationIndex = addressInfo.index + 1;
+        }        
+    }
+    
+    /**
+     *  Returns the first naked address for a given zone.
+     *  If the wallet has no naked addresses, it will generate new addresses until it finds one.
+     */
+    async getAddress(zone: string): Promise<string> {
+        if (!this.validateZone(zone)) throw new Error(`Invalid zone: ${zone}`);
+
+        const shardWalletData = this.#shardWalletsMap.get(zone);
+        if (!shardWalletData) {
+            throw new Error(`Wallet has not been initialized for zone: ${zone}`);
+        }
+        // After the wallet has been initialized, the first naked address is always 
+        // the first address within the pack of last GAP addresses
+        if (shardWalletData.addressesInfo.length < GAP) {
+            throw new Error(`No enough naked addresses available for zone: ${zone}`);
```

Comment:
The comments for this function say "If the wallet has no naked addresses, it will generate new addresses until it finds one." however, an error is thrown if there are not any naked addresses to return. I also noticed that this is an async function that does not use `await` anywhere. Did you mean to generate new addresses within this function until a naked one is found?